server/resource_server/controller/controller_test.go

```diff
@@ -2195,3 +2205,326 @@ func TestRegisterUserHandlerv2_Success(t *testing.T) {
+		"username": "test_user",
+		"signature": "aaabbbbc",
+		"stored_key": "user_stored_key",
+		"server_key": "user_server_key"
+	}`)
+
+	mockService := &MockService{
+		getUserForUsername: func(currUsername string) (models.User, error) {
+			if currUsername != username {
+				t.Fatalf("Username mismatch: expected %s, got %s", "test_user", currUsername)
+			}
+			return models.User{
+				ID:        userId,
+				Username:  username,
+				PublicKey: []byte("mock_public_key"),
+			}, nil
+		},
+		validateSignature: func(message string, signature []byte, publicKey []byte) error {
+			if message != "Sign-in with Layer8" {
+				t.Fatalf("Message mismatch: expected %s, got %s", "Sign-in with Layer8", message)
+			}
+			return nil
+		},
+		updateUserPasswordV2: func(currUsername, currStoredKey, currServerKey string) error {
+			if currUsername != username {
+				t.Fatalf("Username mismatch: expected %s, got %s", username, currUsername)
+			}
+			if currStoredKey != storedKey {
+				t.Fatalf("currStoredKey mismatch: expected %s, got %s", storedKey, currStoredKey)
+			}
+			if currServerKey != serverKey {
+				t.Fatalf("currServerKey mismatch: expected %s, got %s", serverKey, currServerKey)
+			}
+			return nil
+		},
+	}
+	req := httptest.NewRequest("POST", "/api/v2/reset-password", bytes.NewBuffer(reqBody))
+	req = req.WithContext(context.WithValue(req.Context(), "service", mockService))
+	req.Header.Set("Content-Type", "application/json")
+
+	rr := httptest.NewRecorder()
+
+	Ctl.ResetPasswordHandlerV2(rr, req)
+
+	assert.Equal(t, http.StatusCreated, rr.Code)
+
+	response := decodeResponseBodyForResponse(t, rr)
+
+	assert.Equal(t, true, response.IsSuccess)
+	assert.Equal(t, "Your password was updated successfully!", response.Message)
+	assert.Nil(t, response.Error)
+}
+
+func TestResetPasswordHandlerV2_InvalidJSON(t *testing.T) {
+	reqBody := []byte(`{"invalid-request"}`)
+
+	mockService := &MockService{
+		getUserForUsername: func(currUsername string) (models.User, error) {
+			if currUsername != username {
+				t.Fatalf("Username mismatch: expected %s, got %s", username, currUsername)
+			}
+			return models.User{
+				ID:        userId,
+				Username:  username,
+				PublicKey: []byte("mock_public_key"),
+			}, nil
+		},
+		validateSignature: func(message string, signature []byte, publicKey []byte) error {
+			if message != "Sign-in with Layer8" {
+				t.Fatalf("Message mismatch: expected %s, got %s", "Sign-in with Layer8", message)
+			}
+			return nil
+		},
+		updateUserPasswordV2: func(currUsername, currStoredKey, currServerKey string) error {
+			if currUsername != username {
+				t.Fatalf("Username mismatch: expected %s, got %s", username, currUsername)
+			}
+			if currStoredKey != storedKey {
+				t.Fatalf("StoredKey mismatch: expected %s, got %s", storedKey, currStoredKey)
+			}
+			if currServerKey != serverKey {
+				t.Fatalf("ServerKey mismatch: expected %s, got %s", serverKey, currServerKey)
+			}
+			return nil
+		},
+	}
```

Comment:
You can make it just `&MockService{}` because service methods should not get called anyway (since the request is invalid)